scipy/sparse/linalg/_isolve/tests/test_lgmres.py

```diff
@@ -24,29 +25,37 @@
                       [0, 3, 0, 1, -2, 1],
                       [1, 0, 0, 0, 1, -2]]))
 b = array([1, 2, 3, 4, 5, 6])
-count = [0]
-niter = [0]
+count = threading.local()  # [0]
+niter = threading.local()  # [0]
 
 
 def matvec(v):
-    count[0] += 1
+    if not hasattr(count, 'c'):
+        count.c = [0]
+    count.c[0] += 1
     return Am@v
 
 
 def cb(v):
-    niter[0] += 1
+    if not hasattr(niter, 'n'):
+        niter.n = [0]
+    niter.n[0] += 1
 
 
 A = LinearOperator(matvec=matvec, shape=Am.shape, dtype=Am.dtype)
 
 
 def do_solve(**kw):
-    count[0] = 0
+    if not hasattr(niter, 'n'):
+        niter.n = [0]
+    if not hasattr(count, 'c'):
+        count.c = [0]
+    count.c[0] = 0
```

Comment:
if `sparse` expert passes by may want to skim this over?

Reply:
This is a general pattern of using a `threading.local()` variable rather than a module-global one, so looks fine.